airflow/jobs/base_job.py

```diff
@@ -175,6 +176,17 @@ def kill(self, session=None):
     def on_kill(self):
         """Will be called when an external kill command is received."""
 
+    @retry_db_transaction
+    def handle_db_transaction_with_session(self, task_function, session):
+        return task_function(session)
```

Comment:
Why does this need to be defined on the object, instead of a nested local function?

Reply:
I intended it to be accessible from subclasses.

Reply:
I intended ```handle_db_task``` to be accessible from subclasses.
I agree that ```handle_db_transaction_with_session``` could go into ```handle_db_task```.